### ISSUE TITLE
Blank white dialog box put up by Docker Desktop

### ISSUE THREAD

**Issue by rfay** [docker/for-mac#6134]
- [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: E0801D1C-C895-4C7E-A3A1-1B030AACACEA/20220116012419

### Expected behavior

No weird blank white windows

### Actual behavior

I regularly see a weird blank white window. I have just been closing it. Thought it might be an artifact of remote controlling test runner. But no, it's real:

<img width="779" alt="Screen Shot 2022-01-15 at 6 22 26 PM" src="https://user-images.githubusercontent.com/112444/149643847-becabdff-a25d-40f3-8503-79a52dd0741e.png">

### Information

Happens regularly, but not sure why or when. I always thought it was trying to tell me something, but guess I don't know what it's trying to say. 

  - macOS Version: Monterrey
  - Intel chip or Apple chip: It happens regularly on both.
  - Docker Desktop Version: 4.3.2 through 4.5.0 at least

### Output of `/Applications/Docker.app/Contents/MacOS/com.docker.diagnose check`

This diagnostic doesn't work, so you should fix it. Is there an issue for that I wonder.

```
% /Applications/Docker.app/Contents/MacOS/com.docker.diagnose check
Starting diagnostics

[PASS] DD0027: is there available disk space on the host?
[PASS] DD0028: is there available VM disk space?
[PASS] DD0031: does the Docker API work?
[PASS] DD0004: is the Docker engine running?
[PASS] DD0011: are the LinuxKit services running?
[FAIL] DD0016: is the LinuxKit VM running? vm is not running: vm has not started
[PASS] DD0001: is the application running?
[PASS] DD0018: does the host support virtualization?
[FAIL] DD0017: can a VM be started? vm has not started: vm has not started
[PASS] DD0015: are the binary symlinks installed?
[PASS] DD0003: is the Docker CLI working?
[PASS] DD0013: is the $PATH ok?
[PASS] DD0007: is the backend responding?
[PASS] DD0014: are the backend processes running?
[PASS] DD0008: is the native API responding?
[PASS] DD0009: is the vpnkit API responding?
[PASS] DD0010: is the Docker API proxy responding?
[PASS] DD0012: is the VM networking working?
[PASS] DD0032: do Docker networks overlap with host IPs?
[SKIP] DD0030: is the image access management authorized?
[PASS] DD0019: is the com.docker.vmnetd process responding?
[PASS] DD0033: does the host have Internet access?

Please investigate the following 1 issue:

1 : The test: can a VM be started?
    Failed with: vm has not started: vm has not started

The Docker engine runs inside a Linux VM. Therefore we must be able to start Virtual Machines.
```

### Steps to reproduce the behavior

Don't know.

**Comment by igoooor** [comment]
I have the same as you, same Apple chip, same diagnose output

**Comment by rfay** [comment]
I'm wondering if this white empty dialog box is related to other bugs. It seems to be there when I go to restart docker because it's hung, as describe in #6114

**Comment by rfay** [comment]
I continue to see this regularly. I almost always see it when docker has hung, but that just happens to be mostly when I have to restart docker per #6114.  What is this mysterious dialog box?

**Comment by QuinnyPig** [comment]
Seeing this as well. Let's get some answers!

**Comment by arichtman** [comment]
Same on 4.6.1 (76265) though it appears to respect the system appearance settings

**Comment by QuinnyPig** [comment]
Hmm. Data point: I’m not logged in to the app when this happens.

**Comment by trungutt** [comment]
Could you please upload a more recent diagnostic so we could get into this problem? Thanks

**Comment by mark-dr** [comment]
Heya, I work with Trung on the Docker Desktop UI, and we're keen to figure this out for you (thanks for reporting it!). We think it's a combination of two things:

1. some unanticipated situation on your machines throwing an error
2. that error hitting a gap in our handling and causing a full UI crash

We've got some ideas for addressing point 2 that we'll be looking into separately, but looking at diagnostic information is our best shot for point 1. We know you've already provided some, but we delete older diagnostics automatically in case they contain personally identifying information, and we didn't get to this one in time - our fault, sorry about that. If you can provide another batch, we'll look into it ASAP.

**Comment by rfay** [comment]
Here you go, 4D073710-26E7-4DA2-9AA5-BFC1A7A6397E/20220323120333
It happens just all the time. Note that the white dialog went away when I clicked the whale to get the diagnostic.

Nice to see diagnostic upload working again, and even nicer that you're looking at them.

**Comment by mark-dr** [comment]
Thanks! Does Docker Desktop seem to work fine in all other respects? e.g. you can open the dashboard fine, the various options in the whale menu all open functioning pages? Does the window appear even when you don't have any other UI open? Does it seem to correlate with whether any containers are running?

**Comment by rfay** [comment]
The rogue window often appears when there's nothing else open. And I often see it when having to diagnose other problems, like https://github.com/docker/for-mac/issues/6114

I run several macs with Docker Desktop as test runners for [ddev](https://github.com/drud/ddev).  So when they hang like in https://github.com/docker/for-mac/issues/6114 or otherwise misbehave, I connect to them via a remote desktop app and fix things up. I often see the rogue window there. But of course... it's probably showing on other machines that aren't misbehaving. I am pretty sure I've seen it when docker is not broken. For example, in https://github.com/docker/for-mac/issues/6134#issuecomment-1076304829 I didn't have any reason to believe docker was broken, I just went to a couple of the machines to check, and on the second one I found it showing.

**Comment by trungutt** [comment]
We're trying to understand and fix this issue. @rfay @QuinnyPig @arichtman Would you mind try out this build to see if it improves the situation?

Mac Intel: https://desktop-stage.docker.com/mac/main/amd64/76484/Docker.dmg
Mac M1: https://desktop-stage.docker.com/mac/main/arm64/76484/Docker.dmg

**Comment by QuinnyPig** [comment]
Ooh. This appears to be working well after a couple of tests. Now the only useless annoying window devoid of content that's popping up to annoy me is Amazon Chime. Nice work!

**Comment by ebriney** [comment]
Fixed in 4.7.0, the popup is a warning because you mount home dir in a container and so because of fs watchers / event injection it can lead to performance degradation. @rfay let's me know if it's blocking your CI.
